### PR TITLE
VACMS-17515: Hiding service options and appointments on Non-clinical services

### PR DESCRIPTION
**Pull request by omahane**
## Description

Relates to #17515

## Testing done
Manually

## Screenshots
Screenshot 2024-05-22 at 12 23 29 AM

### Set up QA Content Publisher
- [x] As an admin, update the QA Content Publisher
  - [x] Assign the following roles
    - [x] Content creator - VAMC
    - [x] Content creator - VBA
    - [x] Content publisher
  - [x] Assign the following Section
    - [x] VA Boston health care
    - [x] Cheyenne VA Regional Benefit Office

### Add and Edit a VAMC Facility Non-clinical Service
- [x] As QA Content Publisher, [create a VAMC Facility Non-clinical Service](https://pr18169-vrod75xjthebqtv2eicrbqjwtmuy5ulr.ci.cms.va.gov/node/add/vha_facility_nonclinical_service) (Billing and insurance at West Roxbury VA Medical Center)
- [x] Confirm that Service options is not showing in Service locations
- [x] Confirm that Appointments is not showing in Service locations
- [x] Save the node
- [x] Edit the node
- [x] Confirm that Service options is not showing in Service locations
- [x] Confirm that Appointments is not showing in Service locations

### Add and Edit a VAMC Facility Health Service
- [x] As QA Content Publisher, create a VAMC Facility Health Service (Endocrinology at Brockton VA Medical Center)
- [x] Confirm that Service options is showing in Service locations
- [x] Confirm that Appointments is showing in Service locations
- [x] Save the node
- [x] Edit the node
- [x] Confirm that Service options is showing in Service locations
- [x] Confirm that Appointments is showing in Service locations

### Add and Edit a VBA Facility Service
- [x] As QA Content Publisher, create a VBA Facility Service (any)
- [x] Confirm that Service options is showing in Service locations
- [x] Confirm that Appointments is showing in Service locations
- [x] Save the node
- [x] Edit the node
- [x] Confirm that Service options is showing in Service locations
- [x] Confirm that Appointments is showing in Service locations

### Add and Edit a Service Region
- [x] As a content admin, create a Service Region (any)
- [x] Confirm that Service options is showing in Service locations
- [x] Confirm that Appointments is showing in Service locations
- [x] Save the node
- [x] Edit the node
- [x] Confirm that Service options is showing in Service locations
- [x] Confirm that Appointments is showing in Service locations

### Definition of Done

- [ ] Documentation has been updated, if applicable.
- [ ] Tests have been added if necessary.
- [ ] Automated tests have passed.
- [ ] Code Quality Tests have passed.
- [ ] Acceptance Criteria in related issue are met.
- [ ] Manual Code Review Approved.
- [ ] If there are field changes, front end output has been thoroughly checked.

### Select Team for PR review

- [ ] `CMS Team`
- [ ] `Public websites`
- [x] `Facilities`
- [ ] `User support`
- [ ] `Accelerated Publishing`

### CODE REVIEW

File: docroot/modules/custom/va_gov_vamc/va_gov_vamc.module [hunk 1]
```diff
@@ -238,3 +241,36 @@ function va_gov_vamc_entity_bundle_field_info_alter(&$fields, EntityTypeInterfac
...
+ *
+ * @param array $form
+ *   The node form.
+ */
+function _va_gov_vamc_remove_service_location_fields_from_non_clinical_services(&$form) {
+  // If the field_service_location is not found, return.
+  if (!array_key_exists('field_service_location', $form)) {
+    return;
+  }
+  $service_location_widget = $form['field_service_location']['widget'];
```
Comment (dsasser) [inline docroot/modules/custom/va_gov_vamc/va_gov_vamc.module:256]: COULD:

This probably has very low likeliness to break, but this has potential to error if the 'widget' value is not set on the $form['field_service_location'], if $form['field_service_location'] is not iterable, or in the next line, if the 'widget' is not iterable (eg an array).

File: docroot/modules/custom/va_gov_vamc/va_gov_vamc.module [hunk 2]
```diff
@@ -238,3 +241,36 @@ function va_gov_vamc_entity_bundle_field_info_alter(&$fields, EntityTypeInterfac
...
+  foreach ($service_location_widget as $key => $widget) {
+    if (is_numeric($key)) {
+      // Service options and Appointments are not relevant
+      // to non-clinical services.
+      $groups_to_clear = [
+        'group_service_options',
+        'group_appointments',
+      ];
+      foreach ($groups_to_clear as $group) {
+        if (isset($widget['subform']['#fieldgroups'][$group])) {
```
Comment (dsasser) [inline docroot/modules/custom/va_gov_vamc/va_gov_vamc.module:266]: COULD:

This has potential to error If 'subform' is not set on the $widget, or '#fieldgroups' is not set on the 'subform'. You could replace isset() here with !empty().

**Review by dsasser — APPROVED** [review]
Approved, with some comments regarding optional defensive measures to avoid errors in very unlikely scenarios.

File: docroot/modules/custom/va_gov_vamc/va_gov_vamc.module (hunk 2 above)
Comment (omahane) [reply docroot/modules/custom/va_gov_vamc/va_gov_vamc.module:266]: In the Debug Console, I set `$widget = null` before the "if" statement and didn't see an error.

The only effect was that the fields rendered.

**Review by dsasser — APPROVED** [review]
I went through the QA steps and didn't have any issues. Code looks good!

**Pull request merged**